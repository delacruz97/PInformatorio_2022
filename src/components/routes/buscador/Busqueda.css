.contenedor{
    text-align: center;
}
.input{
 
    width: 400px;
    height: 60px;
    border: 1px solid #ccc;
    border-radius: 5px;
    padding: 0 10px;
    font-size: 16px;
    color: #666;
    outline: none;


}
.btn{
    width: 90px;
    height: 50px;
    color: rgba(255, 255, 255, 0.589);
    background-color:rgb(81, 123, 212);
    border: solid 1px #00bbd43d;
    border-radius: 5px;
    margin: 10px auto;
}