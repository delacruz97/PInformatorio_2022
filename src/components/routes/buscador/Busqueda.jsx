import React, { useState } from "react";
import { Link } from "react-router-dom";
import "./Busqueda.css";
import "../../Header.css";
import LogoLink from "../../LogoLink";

/* import axios from "axios"; */

const Busqueda = () => {
  const [noticias, setNoticias] = useState([]);
  const [busqueda, setBusqueda] = useState("");
  const [cargando, setCargando] = useState(false);
  const [error, setError] = useState(false);
  /*   const [cargando, setCargando] = useState(false); */

  /* useEffect(() => {
    const consultarAPI = async () => {
      const url = `https://newsapi.org/v2/everything?q=${busqueda}&apiKey=d4af2b7365e74b1f9f4af4d354f79038&language=es`;
      const resultado = await axios.get(url);
      setNoticias(resultado.data.articles);
    };
    consultarAPI();
  }, [busqueda]); */

  const datos = (e) => {
    e.preventDefault();

    fetch(
      `https://newsapi.org/v2/everything?q=${busqueda}&apiKey=d4af2b7365e74b1f9f4af4d354f79038&page=1&pageSize=10&language=es`
    )
      .then((response) => response.json())
      .then((data) => {
        const { articles } = data;
        const dataN = articles || [];
        console.log(data);
        setNoticias(articles);

        if (dataN.length === 0) {
          setError(true);
        } else {
          setError(false);
        }
      });
  };

  console.log(noticias);

  const handleChange = (e) => {
    setBusqueda(e.target.value);
    if (busqueda.length >= 3) {
      setCargando(true);
      console.log(cargando);
    } else {
      setCargando(false);
      console.log(cargando);
    }
  };

  return (
    <>
      <LogoLink />
      <form onSubmit={datos}>
        <div className="contenedor">
          {error === true ? (
            <p className="error">No hay resultados</p>
          ) : (
            console.log("exitoso")
          )}
          <Link to="/buscador">
            <input
              className="input"
              type="search"
              placeholder="Buscar"
              onChange={handleChange}
            />
          </Link>

          <br></br>

          {cargando === true ? (
            <button className="btn" type="submit">
              Buscar
            </button>
          ) : null}
        </div>
      </form>
      <div className="cont_principal">
        {noticias.map((user) => (
          <div key={user.title} className="container">
            <div>
              <p className="name">{user.source.name}</p>
              <a href={user.url} target="blank" className="sumbrayado">
                <h1 className="titulo">{user.title}</h1>
              </a>
              <p className="description">{user.description}</p>
              <p className="fecha">{user.publishedAt}</p>
            </div>

            <div>
              <img className="img" src={user.urlToImage} alt={user.title} />
            </div>
          </div>
        ))}
      </div>
    </>
  );
};

export default Busqueda;

//
