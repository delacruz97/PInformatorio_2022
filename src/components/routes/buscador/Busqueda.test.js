import { screen, render } from "@testing-library/react";

import Busqueda from "./Busqueda";

describe("Busqueda", () => {
  it("etiqueta p", () => {
    render(<Busqueda />);
    expect(screen.getByText(/No hay resultados/i)).toBeInTheDocument();
  });
});
