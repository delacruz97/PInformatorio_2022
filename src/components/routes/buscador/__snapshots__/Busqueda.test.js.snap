// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Busqueda Debe de retornar un componente 1`] = `<Busqueda />`;
